components/browser/session-storage/src/androidTest/java/RestoringBrowsingSessionsTest.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import mozilla.components.browser.engine.gecko.GeckoEngine
+import mozilla.components.browser.session.storage.SessionStorage
+import mozilla.components.browser.session.storage.getFileForEngine
+import mozilla.components.support.ktx.util.writeString
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * The test cases in this class restore actual browsing sessions taken from devices. If a test case
+ * breaks then this is a strong indication that the restore code changed in a non backwards compatible
+ * way and that users may lose data.
+ *
+ * The tests here do also restore the "engine" state using GeckoView Nightly. Breaking changes in
+ * GeckoView may also cause session restore to break in a non backwards compatible way. Such breakages
+ * need to be reported to the GeckoView team to be fixed before reaching release branches.
+ */
+@Suppress("TestFunctionName")
+@RunWith(AndroidJUnit4::class)
+class RestoringBrowsingSessionsTest {
+    private val context: Context
+        get() = ApplicationProvider.getApplicationContext()
+
+    /**
+     * App: Sample Browser
+     * Version: 2020-12-16
+     * GeckoView: 84.0 Beta (3147bf02ca678985f814ea2b091a194a9ac22f71)
+     */
```

Comment:
I want to add some more cases here and grab some serialized states from different versions of Fenix.

Reply:
🔥 awesome to have those tests.